lib/resolvers/project-version.js

```diff
@@ -92,21 +92,28 @@ module.exports = ({ models }) => ({ action, data, id }, transaction) => {
   }
 
   if (action === 'patch') {
+    const version = await ProjectVersion.query(transaction).findById(id);
+    const project = await Project.query(transaction).findById(version.projectId);
+
     return Promise.resolve()
-      .then(() => ProjectVersion.query(transaction).findById(id))
-      .then(version => {
+      .then(() => {
         version.data = version.data || {};
         version.data.protocols = version.data.protocols || [];
         try {
           const newData = jsondiff.patch(version.data, data.patch);
 
-          return version.$query(transaction).patchAndFetch({
+          const patch = {
             data: {
               ...newData,
               retrospectiveAssessment: retrospectiveAssessment.addedByAsru(newData)
-            },
-            raCompulsory: retrospectiveAssessment.isRequired(newData)
-          });
+            }
+          };
+
+          if (project.status === 'inactive') {
+            patch.raCompulsory = retrospectiveAssessment.isRequired(newData);
```

Comment:
This isn't quite sufficient because it won't handle the case where a project doesn't need RA but is amended so it does.

There needs to be an `else` that gets the RA state of the granted version and does a:

```
path.raCompulsory = grantedRaCompulsory || retrospectiveAssessment.isRequired(newData);
```

Reply:
right you are. I need to stop trying to developer